tests/functional/test_credentials.py

```diff
@@ -826,7 +826,7 @@ def test_credential_process_returns_error(self):
             # Finally `(?s)` at the beginning makes dots match newlines so
             # we can handle a multi-line string.
             reg = r"(?s)^((?!b').)*$"
-            with self.assertRaisesRegexp(CredentialRetrievalError, reg):
+            with six.assertRaisesRegex(self, CredentialRetrievalError, reg):
```

Comment:
Let's use [`pytest.raises`](https://docs.pytest.org/en/stable/reference.html#pytest-raises), and we can pass the regex as the `match` parameter. That'll avoid us needing to import six here.